pds4indextools/pds4_create_xml_index.py

```diff
@@ -198,271 +701,781 @@ def update_nillable_elements_from_xsd_file(xsd_file, nillable_elements_info):
-                             '.ini file.')
-
-    args = parser.parse_args()
+        clean_headers(df)
+
+    if args.fixed_width:
+        padded_df = pad_column_values_and_headers(df)
+        padded_df.to_csv(output_csv_path, index=False, na_rep='NaN')
+
+    else:
+        df.to_csv(output_csv_path, index=False, na_rep='NaN')
+
+
+def find_base_attribute(xsd_tree, target_name, new_namespaces):
+    """
+    Finds the base attribute of a target element in an XML schema.
+
+    This function searches for the base type of a given target element in the provided
+    XML schema tree. It follows nested type definitions to return the final meaningful
+    base type, such as `pds:ASCII_NonNegative_Integer`.
+
+    Args:
+        xsd_tree (etree._Element): The XML schema tree.
+        target_name (str): The name of the target element to search for.
+
+    Returns:
+        str: The base type of the target element if found, otherwise None.
+
+    Raises:
+        etree.XPathEvalError: If there is an error during XPath evaluation.
+    """
+
+    # Register namespaces
+    namespaces = {
+        'xs': 'http://www.w3.org/2001/XMLSchema',
+        'pds': 'http://pds.nasa.gov/pds4/pds/v1'
+    }
+    namespaces.update(new_namespaces)
+
+    def follow_base_type(base_type):
+        """
+        Recursively follows the base type definitions to find the final base type.
+
+        Args:
+            base_type (str): The initial base type to follow.
+
+        Returns:
+            str: The final base type.
+        """
+        while True:
+            if 'ASCII' in base_type or 'UTF8' in base_type:
+                return base_type
+
+            next_query = (
+                f".//xs:simpleType[@name='{base_type.split(':')[-1]}']"
+                f"//xs:restriction/@base"
+            )
+            try:
+                next_result = xsd_tree.xpath(next_query, namespaces=namespaces)
+            except Exception:
+                break
+            if not next_result:
+                break
+            base_type = next_result[0]
+        return base_type
+
+    def get_base_type(query):
+        """
+        Executes an XPath query to find the base type.
+
+        Args:
+            query (str): The XPath query to execute.
+
+        Returns:
+            list: The result of the XPath query.
+        """
+        try:
+            result = xsd_tree.xpath(query, namespaces=namespaces)
+            return result
+        except etree.XPathEvalError:
+            return None
+
+    # Define the initial XPath query to find the complexType element by name
+    initial_query = f".//xs:complexType[@name='{target_name}']//xs:extension/@base"
+    initial_result = get_base_type(initial_query)
+
+    if initial_result:
+        base_type = initial_result[0]
+        return follow_base_type(base_type)
+
+    # If the initial query did not find a result, try other patterns
+    other_queries = [
```

Comment:
Why are these separated out? Can't you just put the initial query first in this list?